kolibri/tasks/api.py

```diff
@@ -1,84 +1,140 @@
 
     def list(self, request):
-        return Response(get_tasks())
+        from django_q.models import Task
+        tasks_response = [_task_to_response(t) for t in Task.objects.all()]
+        return Response(tasks_response)
 
     def create(self, request):
-        data = request.data
-
-        task_id = schedule_command(data["command"], *data["args"], **data["kwargs"])
-
-        return Response(task_id)
+        # unimplemented. Call out to the task-specific APIs for now.
+        pass
 
     def retrieve(self, request, pk=None):
-        if pk:
-            return Response(get_task_state(pk))
+        from django_q.models import Task
+
+        task = _task_to_response(Task.get_task(pk))
+        return Response(task)
 
     def destroy(self, request, pk=None):
-        if pk:
-            return Response(cancel_task(pk))
+        # unimplemented for now.
+        pass
+
+    # convenience functions for triggering often used tasks
+
+    @list_route(methods=['post'])
+    def startremoteimport(self, request):
+        '''Download a channel's database from the main curation server, and then
+        download its content.
+
+        '''
+        TASKTYPE = "remoteimport"
+
+        # loading django_q apparently requires django settings to load first.
+        # Import here to avoid circular imports.
+        from django_q.tasks import async
+        from django_q.models import Task
+        data = json.loads(request.body.decode('utf-8'))
+
+        if "id" not in data:
+            raise serializers.ValidationError("The 'id' field is required.")
+
+        channel_id = data['id']
+
+        task_id = async(_importchannel, channel_id, group=TASKTYPE, progress_updates=True)
+
+        # id status metadata
+
+        # wait for the task instance to be saved first before continuing
+        taskobj = Task.get_task(task_id)
+        if taskobj:             # the task object has been saved!
+            resp = _task_to_response(taskobj)
+        else:                   # task object hasn't been saved yet, fake the response for now
+            resp = {
+                "type": TASKTYPE,
+                "status": "PENDING",
+                "percentage": 0,
+                "metadata": {},
+                "id": task_id,
+            }
+
+        return Response(resp)
+
+    @list_route(methods=['post'])
+    def startlocalimportchannel(self, request):
+        '''
+        Import a channel locally, and move content to the local machine.
+
+        '''
+
+    @list_route(methods=['get'])
+    def localdrive(self, request):
+        drives = get_mounted_drives_with_channel_info()
+
+        # make sure everything is a dict, before converting to JSON
+        assert isinstance(drives, dict)
+
+        out = []
+        for mountdata in drives.values():
+            mountdata = mountdata._asdict()
+            if mountdata['metadata']['channels']:
+                mountdata['channels'] = [c._asdict() for c in mountdata['channels']]
```

Comment:
`namedtuple` is pretty simple -- it has a `_asdict()` function which turns it into a dict. However, the transformations here are made to conform to the schema that we @indirectlylit, @ralphiee22 agreed to! So in this case it's endpoint specific.